compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -724,11 +728,11 @@ impl<'context> Elaborator<'context> {
     }
 
     fn find_numeric_generics(
-        parameters: &[Type],
+        parameters: &Parameters,
         return_type: &Type,
     ) -> Vec<(String, TypeVariable)> {
         let mut found = BTreeMap::new();
-        for parameter in parameters {
+        for (_, parameter, _) in &parameters.0 {
```

Comment:
It looks like we are only using the type of parameter anyway here. Perhaps we keep this as accepting `[Type]`? Probably could have a method on `Parameters` to extract the param types.